services/121-service/src/payments/payments.service.ts

```diff
@@ -714,6 +733,86 @@ export class PaymentsService {
     );
   }
 
+  /**
+   * Creates and adds safaricom transaction jobs.
+   *
+   * This method is responsible for creating transaction jobs for Safaricom. It fetches necessary PA data and maps it to a FSP specific DTO.
+   * It then adds these jobs to the transaction queue.
+   *
+   * @returns {Promise<void>} A promise that resolves when the transaction jobs have been created and added.
+   *
+   */
+  private async createAndAddSafaricomTransactionJobs({
+    referenceIdsAndTransactionAmounts: referenceIdsTransactionAmounts,
+    programId,
+    userId,
+    paymentNumber,
+    isRetry,
+  }: {
+    referenceIdsAndTransactionAmounts: ReferenceIdAndTransactionAmountInterface[];
+    programId: number;
+    userId: number;
+    paymentNumber: number;
+    isRetry: boolean;
+  }): Promise<void> {
+    const safaricomQuestions =
+      await this.financialServiceProviderQuestionRepository.getQuestionsByFspName(
+        FinancialServiceProviderName.safaricom,
+      );
+    const safaricomQuestionNames = safaricomQuestions.map((q) => q.name);
```

Comment:
Much of the code is (almost) identical to what is in createAndAddIntersolveVisaTransactionJobs() => refactor this out into private functions (keep SRP in mind), so duplicate code is minimized.